File: Data Structures/C/Stack.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

struct Stack {
    int arr[MAX_SIZE];
    int top;
};

typedef struct Stack Stack;

void init(Stack* stack) {
    stack->top = -1;
}

int is_empty(Stack* stack) {
    return stack->top == -1;
}

int is_full(Stack* stack) {
    return stack->top == MAX_SIZE - 1;
}

void push(Stack* stack, int item) {
    if (!is_full(stack)) {
        stack->arr[++stack->top] = item;
    } else {
        printf("Stack overflow\n");
    }
}

int pop(Stack* stack) {
    if (!is_empty(stack)) {
        return stack->arr[stack->top--];
    } else {
        printf("Stack underflow\n");
        return -1;
    }
}

int peek(Stack* stack) {
    if (!is_empty(stack)) {
        return stack->arr[stack->top];
    } else {
        printf("Stack is empty\n");
        return -1;
    }
}

int main() {
    Stack stack;
    init(&stack);

    push(&stack, 10);
    push(&stack, 20);
    push(&stack, 30);

    printf("Top element: %d\n", peek(&stack));

    printf("Popped element: %d\n", pop(&stack));
    printf("Popped element: %d\n", pop(&stack));

    printf("Top element after pops: %d\n", peek(&stack));

    return 0;
}

File: Data Structures/Java/Stack.java
import java.util.EmptyStackException;

class Stack {
    private static final int MAX_SIZE = 100;
    private int[] arr;
    private int top;

    public Stack() {
        arr = new int[MAX_SIZE];
        top = -1;
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == MAX_SIZE - 1;
    }

    public void push(int item) {
        if (!isFull()) {
            arr[++top] = item;
        } else {
            System.out.println("Stack overflow");
        }
    }

    public int pop() {
        if (!isEmpty()) {
            return arr[top--];
        } else {
            throw new EmptyStackException();
        }
    }

    public int peek() {
        if (!isEmpty()) {
            return arr[top];
        } else {
            throw new EmptyStackException();
        }
    }

    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(10);
        stack.push(20);
        stack.push(30);

        System.out.println("Top element: " + stack.peek());

        System.out.println("Popped element: " + stack.pop());
        System.out.println("Popped element: " + stack.pop());

        System.out.println("Top element after pops: " + stack.peek());
    }
}
